File: chat-uikit/src/main/java/io/agora/chat/uikit/chat/viewholder/EaseImageViewHolder.java
package io.agora.chat.uikit.chat.viewholder;

import android.net.Uri;
import android.view.View;

import androidx.annotation.NonNull;

import io.agora.chat.ChatClient;
import io.agora.chat.ChatMessage;
import io.agora.chat.ImageMessageBody;
import io.agora.chat.uikit.manager.EaseActivityProviderHelper;
import io.agora.chat.uikit.manager.EaseConfigsManager;
import io.agora.chat.uikit.utils.EaseFileUtils;
import io.agora.util.EMLog;


public class EaseImageViewHolder extends EaseChatRowViewHolder {

    public EaseImageViewHolder(@NonNull View itemView) {
        super(itemView);
    }

    @Override
    public void onBubbleClick(ChatMessage message) {
        super.onBubbleClick(message);
        ImageMessageBody imgBody = (ImageMessageBody) message.getBody();

        if(!EaseConfigsManager.enableSendChannelAck()) {
            //Here no longer send read_ack message separately, instead enter the chat page to send channel_ack
            //New messages are sent in the onReceiveMessage method of the chat page, except for video
            // , voice and file messages, and send read_ack messages
            if (message != null && message.direct() == ChatMessage.Direct.RECEIVE && !message.isAcked()
                    && message.getChatType() == ChatMessage.ChatType.Chat) {
                try {
                    ChatClient.getInstance().chatManager().ackMessageRead(message.getFrom(), message.getMsgId());
                } catch (Exception e) {
                    e.printStackTrace();
                }
            }
        }
        Uri imgUri = imgBody.getLocalUri();
        EaseFileUtils.takePersistableUriPermission(getContext(), imgUri);
        EMLog.e("Tag", "big image uri: " + imgUri + "  exist: "+EaseFileUtils.isFileExistByUri(getContext(), imgUri));
        if(EaseFileUtils.isFileExistByUri(getContext(), imgUri)) {
            EaseActivityProviderHelper.startToLocalImageActivity(getContext(), imgUri);
        } else{
            // The local full size pic does not exist yet.
            // ShowBigImage needs to download it from the server
            // first
            EaseActivityProviderHelper.startToLocalImageActivity(getContext(), message.getMsgId(), imgBody.getFileName());
        }
    }

    @Override
    protected void handleReceiveMessage(ChatMessage message) {
        super.handleReceiveMessage(message);
        getChatRow().updateView(message);
    }
}

File: ease-im-kit/src/main/kotlin/com/hyphenate/easeui/feature/chat/viewholders/EaseImageViewHolder.kt
package com.hyphenate.easeui.feature.chat.viewholders

import android.net.Uri
import android.view.View
import com.hyphenate.easeui.EaseIM
import com.hyphenate.easeui.feature.chat.activities.EaseShowBigImageActivity
import com.hyphenate.easeui.common.ChatClient
import com.hyphenate.easeui.common.ChatImageMessageBody
import com.hyphenate.easeui.common.ChatLog
import com.hyphenate.easeui.common.ChatMessage
import com.hyphenate.easeui.common.ChatMessageDirection
import com.hyphenate.easeui.common.ChatType
import com.hyphenate.easeui.common.utils.EaseFileUtils

class EaseImageViewHolder(itemView: View) : EaseChatRowViewHolder(itemView) {
    override fun onBubbleClick(message: ChatMessage?) {
        super.onBubbleClick(message)
        val imgBody: ChatImageMessageBody? = message?.body as? ChatImageMessageBody
        if (EaseIM.getConfig()?.chatConfig?.enableSendChannelAck == false) {
            //Here no longer send read_ack message separately, instead enter the chat page to send channel_ack
            //New messages are sent in the onReceiveMessage method of the chat page, except for video
            // , voice and file messages, and send read_ack messages
            message?.let {
                if (it.direct() === ChatMessageDirection.RECEIVE && !it.isAcked && it.chatType === ChatType.Chat) {
                    try {
                        ChatClient.getInstance().chatManager()
                            .ackMessageRead(it.from, it.msgId)
                    } catch (e: Exception) {
                        e.printStackTrace()
                    }
                }
            }

        }
        val imgUri: Uri? = imgBody?.getLocalUri()
        EaseFileUtils.takePersistableUriPermission(mContext, imgUri)
        ChatLog.e(
            "Tag",
            "big image uri: " + imgUri + "  exist: " + EaseFileUtils.isFileExistByUri(
                mContext,
                imgUri
            )
        )
        if (EaseFileUtils.isFileExistByUri(mContext, imgUri)) {
            EaseShowBigImageActivity.actionStart(mContext, imgUri)
        } else {
            // The local full size pic does not exist yet.
            // ShowBigImage needs to download it from the server
            // first
            EaseShowBigImageActivity.actionStart(mContext, message?.msgId, imgBody?.fileName)
        }
    }

    override fun handleReceiveMessage(message: ChatMessage?) {
        super.handleReceiveMessage(message)
        getChatRow()?.updateView()
    }
}